src/blueapi/worker/task_worker.py

```diff
@@ -242,38 +291,50 @@ def run(self) -> None:
         self._stopping.clear()
         self._stopped.set()
 
+    @start_as_current_span(TRACER, "defer")
     def pause(self, defer=False):
         LOGGER.info("Requesting to pause the worker")
         self._ctx.run_engine.request_pause(defer)
 
+    @start_as_current_span(TRACER)
     def resume(self):
         LOGGER.info("Requesting to resume the worker")
         self._ctx.run_engine.resume()
 
+    @start_as_current_span(TRACER)
     def _cycle_with_error_handling(self) -> None:
         try:
             self._cycle()
         except Exception as ex:
             self._report_error(ex)
 
+    @start_as_current_span(TRACER)
     def _cycle(self) -> None:
         try:
             LOGGER.info("Awaiting task")
             next_task: TrackableTask | KillSignal = self._task_channel.get()
             if isinstance(next_task, TrackableTask):
                 LOGGER.info(f"Got new task: {next_task}")
                 self._current = next_task  # Informing mypy that the task is not None
+
+                self._current_task_otel_context = get_trace_context()
+                add_span_attributes({"next_task.task_id": next_task.task_id})
+
                 self._current.is_pending = False
                 self._current.task.do_task(self._ctx)
```

Comment:
There are now enough pieces of context that the worker needs to keep track of that it makes sense to box them up into a dataclass or similar object. I think that's a separate issue though...